src/main/java/org/traccar/api/resource/NotificationResource.java

```diff
@@ -113,15 +114,27 @@ public Response testMessage(@PathParam("notificator") String notificator)
     public Response sendMessage(
             @PathParam("notificator") String notificator, @QueryParam("userId") List<Long> userIds,
             NotificationMessage message) throws MessageException, StorageException {
-        permissionsService.checkAdmin(getUserId());
+        permissionsService.checkManager(getUserId());
         List<User> users;
         if (userIds.isEmpty()) {
-            users = storage.getObjects(User.class, new Request(new Columns.All()));
+            users = permissionsService.notAdmin(getUserId()) ?
+                    storage.getObjects(User.class, new Request(new Columns.All(),
+                            new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups())) :
+                    storage.getObjects(User.class, new Request(new Columns.All()));
```

Comment:
Let's do if/else here. Tertiary expression we mostly use for short things that make sense to inline.